src/homepage/howTo/custom-listeners.html

```diff
@@ -0,0 +1,248 @@
+                        That means if there are three listeners registered to the same group, each time an event is fired, only one of these three listeners can receive that event.
+                        You can write your own custom Listener and configure James to register it as a Group Registration.
+                    </li>
+                    <li>
+                        <b>Key Registration</b>: a Key Registration will listen only for events it is interested in, and with each event fired to a Key Registration, it will be
+                        delivered to all registered listeners. That means if there are three listeners registered to the same key, each time an event is fired, all of three listeners will receive that event.
+                        At the moment, James doesn't support to configure custom Listener for Key Registration.
+                        This feature is used internally to implement things like IMAP IDLE aka notifications.
+                        Therefore, it's not exposed through configurations as it makes little sense
+                    </li>
+                </ul>
+
+                <header class="major">
+                    <h2><b>How to use a custom Mailbox Listener in James</b></h2>
+                </header>
+
+                <p>
+                    <b>prerequisite</b>: custom Mailbox Listeners can only work with James Guice products.
+                </p>
+
+                <header class="major">
+                    <h3><b>Use the custom BigMessageListener in James</b></h3>
+                </header>
+
+                <p>
+                    Once you have a custom Listener, it's very simple to setup James with that Listener.
+                    In this example, we will use the
+                    <a href="https://github.com/apache/james-project/blob/master/examples/custom-mailets/src/main/java/org/apache/james/examples/custom/listeners/BigMessageListener.java">BigMessageListener</a>
+                    which will listen events fired after emails are appended to users mailboxes,
+                    then for each email added event, determine the size of added email by getting size information from the event.
+                    If the size of the email is greater than or equals 1 MB, then BigMessageListener will add a "BIG_MESSAGE" flag to that email
+                </p>
+
+                <header class="major">
+                    <h3><b>Starting James with BigMessageListener</b></h3>
+                </header>
+
+                <p>
+                    When you write a custom Listener, you have to compile it and place the compiled jar package inside "extensions-jars" directory
+                    to make James load your Listener when it starts up. But BigMessageListener is already available in James products.
+                    So we can use it directly without that extra step.
```

Comment:
that's unfortunate for a how to because the reader won't learn how to do it

Reply:
oh sorry, that sentence is not correct. we always have to do that step.